zerver/tests/test_bugdown.py

```diff
@@ -892,6 +898,210 @@ def render(msg: Message, content: str) -> str:
+        sender_user_profile = self.example_user('polonius')
+        user_ids.add(sender_user_profile.id)
+        msg = Message(sender=sender_user_profile, sending_client=get_client("test"))
+        realm_alert_words_automaton = get_alert_word_automaton(sender_user_profile.realm)
+
+        def render(msg: Message, content: str) -> str:
+            return render_markdown(msg,
+                                   content,
+                                   realm_alert_words_automaton=realm_alert_words_automaton,
+                                   user_ids = user_ids)
+
+        content = """This is to test out alert words work in languages with accented characters too
+        bonjour est (énormément) ce a quoi ressemble le français
+        et j'espère qu'il n'y n' réglementaire a pas de mots d'alerte dans ce texte français
+        """
+        render(msg, content)
+        expected_user_ids = {user_profiles['hamlet'].id, user_profiles['cordelia'].id}  # type: Set[int]
+        self.assertEqual(msg.user_ids_with_alert_words, expected_user_ids)
+
+    def test_alert_words_returns_empty_user_ids_with_alert_words(self) -> None:
+        alert_words_for_users = {
+            'hamlet': [], 'cordelia': [], 'iago': [], 'prospero': [],
+            'othello': [], 'aaron': []
+        }  # type: Dict[str, List[str]]
+        user_profiles = {}  # type: Dict[str, UserProfile]
+        user_ids = set()  # type: Set[int]
+        for (username, alert_words) in alert_words_for_users.items():
+            user_profile = self.example_user(username)
+            user_profiles.update({username: user_profile})
+            do_set_alert_words(user_profile, alert_words)
+        sender_user_profile = self.example_user('polonius')
+        msg = Message(sender=user_profile, sending_client=get_client("test"))
+        realm_alert_words_automaton = get_alert_word_automaton(sender_user_profile.realm)
+
+        def render(msg: Message, content: str) -> str:
+            return render_markdown(msg,
+                                   content,
+                                   realm_alert_words_automaton=realm_alert_words_automaton,
+                                   user_ids = user_ids)
+
+        content = """hello how is this possible how are you doing today
+        This is to test that the no user_ids who have alrert wourldword is participating
+        in sending of the message
+        """
+        render(msg, content)
+        expected_user_ids = set()  # type: Set[int]
+        self.assertEqual(msg.user_ids_with_alert_words, expected_user_ids)
+
+    def get_random_alert_words(self, num_words: int, word_length: int) -> List[str]:
+        alert_words = []  # type List[str]
+        allowed_characters = string.ascii_letters + string.digits
+        for x in range(num_words):
+            alert_word = ''.join([random.choice(allowed_characters) for i in range(10)])
+            alert_words.append(alert_word)
+        return alert_words
```

Comment:
We don't allow true randomization in our tests, so that we can't have nondeterministic flakes.  Obviously, these random strings are so long that's unlikely to be an issue, but to fit with those standards, can you add a `random.seed(10)` call here?

Reply:
(Or you could just do e.g. `'x' * 500`?)

Reply:
Yes, makes sense. I added this bit so that the huge list of alert_words for profiling resembles the real world usage of alert-words as close as possible.